### DISCUSSION TITLE
About var spp

### DISCUSSION THREAD

**Discussion by wkMvg** [NVlabs/instant-ngp#591] [Q&A]
Hi, 
 instant ngp is really a great work;
 I have question about var spp; what does it mean? samples per pixel? If traditional nerf, samples per ray is about 64 or 128; I read run.py that you set it as 8; So I am confused about it. Can you explain it? Thx a lot

**Comment by Tom94** [comment]
Hi there, spp indeed refers to "samples per pixel", but a "sample" on the image plane is an entire ray. So `spp=8` effectively means that the rendered images have 8x super-sampled anti aliasing, where each of the 8 rays per pixel uses a raymarching step size according to the description in the paper. It's a bit more complicated than taking a fixed number of steps (as original NeRF would): empty space is skipped and occluded regions are culled away.

**Reply by wkMvg** [reply]
Got it! Thx!
